### PR TITLE
docs: Document lint tests

### PR DESCRIPTION
**Pull request by masonicboom**

### CODE REVIEW

File: test/README.md [hunk 1]
```diff
@@ -3,17 +3,18 @@ utilities in their entirety. It does not contain unit tests, which
 can be found in [/src/test](/src/test), [/src/wallet/test](/src/wallet/test),
 etc.
 
-There are currently two sets of tests in this directory:
+There are currently three sets of tests in this directory:
```
Comment (Empact) [inline test/README.md]: nit: how about "the following" - the number isn't particularly important

File: test/README.md [hunk 2]
```diff
@@ -180,6 +181,28 @@ Note: gdb attach step may require `sudo`
...
 
+### Lint tests
+
+#### Dependencies
+
+The lint tests require a python library called flake8. To install it: `pip3 install flake8`.
+
+#### Running the tests
+
+Individual tests can be run by directly calling the test script, eg:
```
Comment (practicalswift) [inline test/README.md]: Nit: `e.g.`

File: test/README.md [hunk 3]
```diff
@@ -180,6 +181,28 @@ Note: gdb attach step may require `sudo`
...
+test/lint/lint-filenames.sh
+```
+
+You can run all the shell-based lint tests by running:
+
+```
+test/lint/lint-all.sh
+```
+
+NOTE: on macOS you may receive warnings like `grep: empty (sub)expression`. To fix, install GNU grep.
```
Comment (practicalswift) [inline test/README.md]: Which scripts trigger this? We should try to fix that.

Comment (practicalswift) [reply test/README.md]: Thanks! Fixed in #13720 :-)

**Comment by MarcoFalke** [comment]
utACK b5f1c28bc7d875681ce5f74ea8e32d601c2557b2, but should wait for the nit

**Comment by laanwj** [comment]
ping @masonicboom, please either update for the nit or reject it

File: test/README.md (hunk 3 above)
Comment (practicalswift) [reply test/README.md]: Please remove this. It is no longer accurate :-)

**Comment by fanquake** [comment]
Replaced by #14526.

**Pull request closed without merging**